lesson11/main.js

```diff
@@ -17,16 +17,20 @@ const loaded = () => {
 
 window.onload = () => {
   loading();
+
   const fetchedData = new Promise((resolve, reject) => {
-    const listItems = [
-      { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
-      { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" },
-    ];
     setTimeout(() => {
-      resolve(listItems);
+      resolve(listItems());
     }, 2500);
   });
```

Comment:
Since we will be using Fetch, I don't think this Promise process is necessary.

Let's think about why we use asynchronous processing.
By delegating heavy processing to asynchronous processing and separating it from the main thread, the goal is to have the JavaScript process run without stopping.

In this case, the heavy processing is to get the json.
This time, fetch will be in charge of executing this heavy processing.

For this reason, does not need the Promise process called fetchedData.

Also, in fetchedData, we are using setTimeout.
In fetchedData, we use setTimeout to make it look like we are actually communicating.
In this case, we are actually fetching JSON using fetch, so there is no need to process setTimeout.

Reply:
Thank you for your lots of advices! 

<What I did>
1. Deleted `Promise` function
2. Changed neme `listItems` to `fetchedData`
3. Implemented it in the try part of try-catch-finally
4. Implemented `jsonUrl` outside of `fetchedData`

Yesterday I rewrote just `listItems` with fetch and I didn't understand what this whole codes work. Actually the code I made until lesson10 was a pseudo-loading screen using setTimeout so now I didn't need it. I didn't think about that and just implemented the code by rewriting the relevant parts.Oh my gosh...
I appriciated your kindly explanation! Please kindly tell me if I wrong again.

Reply:
That's a much cleaner code!✊
It would be great if we could compare and understand what unnecessary processing the code was doing before the fix, and what the code is doing now!

> Actually the code I made until lesson10 was a pseudo-loading screen using setTimeout so now I didn't need it

That's what I did at first, and I think that's what you would write if you don't understand what setTimeout represents. :)